### DISCUSSION TITLE
关于anime4k的通用移植想法

### DISCUSSION THREAD

**Discussion by kuuhaku1314** [Blinue/Magpie#1044] [Q&A]
最近在使用potplayer播放器的时候发现可以使用SVP的AI插帧，在使用madvr渲染后，视频会非常高清流畅
另外potplayer+madvr还支持自定义hlsl的sharder，但不支持glsl

实际上在播放动漫时，anime4k远比madvr的放缩算法更好，但anime4k原仓库hlsl只有0.9版本，glsl已经到4.0版本，许多人苦于这个问题只能选择旧版或者更换为mpv播放器，但mpv是命令行的编码器需要大量配置，对比mpv，potplayer的ui是十分优秀的

我尝试使用Magpie移植的anime4k shader后发现没有任何效果，之后确认原因是shader内使用了许多自定义的MagpieFX拓展
我没有任何hlsl编写经验，但其他语言经验比较丰富，想确认下把anime4k移植到通用的hlsl是否会很困难🤔

如果有大佬能够帮忙移植成通用的shader文件就更好了，感谢🙏🙏

**Comment by Blinue** [comment]
看看 #690。不同软件的着色器不能通用，但既然同为 hlsl，移植应该不会太难。

**Reply by kuuhaku1314** [reply]
好的，我查了下potplayer支持dx9和dx11的shader model
